cli/cmd/proxy/list/command.go

```diff
@@ -211,21 +212,51 @@ func (c *ListCommand) fetchPods() ([]v1.Pod, error) {
 
 	// Fetch all pods in the namespace with labels matching the gateway component names.
 	gatewaypods, err := c.kubernetes.CoreV1().Pods(c.namespace()).List(c.Ctx, metav1.ListOptions{
-		LabelSelector: "component in (ingress-gateway, mesh-gateway, terminating-gateway), chart=consul-helm",
+		LabelSelector: "component in (api-gateway, ingress-gateway, mesh-gateway, terminating-gateway), chart=consul-helm",
 	})
 	if err != nil {
 		return nil, err
 	}
 	pods = append(pods, gatewaypods.Items...)
 
-	// Fetch all pods in the namespace with a label indicating they are an API gateway.
+	// Fetch API Gateway pods with deprecated label and append if they aren't already in the list
+	// TODO this block can be deleted if and when we decide we are ok with no longer listing pods of people using previous API Gateway
+	// versions.
 	apigatewaypods, err := c.kubernetes.CoreV1().Pods(c.namespace()).List(c.Ctx, metav1.ListOptions{
 		LabelSelector: "api-gateway.consul.hashicorp.com/managed=true",
 	})
+
+	namespacedName := func(pod v1.Pod) string {
+		return pod.Namespace + pod.Name
+	}
```

Comment:
Probably fine in this context, but this has the same name as a struct in Kubernetes that is structured somewhat differently. If you wanted to leverage this struct, it might be more "idiomatic". Because it has a `string` method itself, the behavior will be identical when used with formatting or with `string()`.

https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go

Again, not a blocker, just a little note.